libvast/builtins/operators/shell.cpp

```diff
@@ -0,0 +1,209 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/chunk.hpp>
+#include <vast/concept/parseable/string/quoted_string.hpp>
+#include <vast/concept/parseable/vast/pipeline.hpp>
+#include <vast/error.hpp>
+#include <vast/logger.hpp>
+#include <vast/pipeline.hpp>
+#include <vast/plugin.hpp>
+#include <vast/si_literals.hpp>
+
+#include <boost/asio.hpp>
+#include <boost/process.hpp>
+#include <caf/detail/scope_guard.hpp>
+
+#include <mutex>
+#include <queue>
+#include <thread>
+
+namespace vast::plugins::shell {
+namespace {
+
+class shell_operator final : public crtp_operator<shell_operator> {
+public:
+  explicit shell_operator(std::string command) : command_{std::move(command)} {
+  }
+
+  auto operator()(generator<chunk_ptr> input,
+                  operator_control_plane& ctrl) const -> generator<chunk_ptr> {
```

Comment:
I was hoping to be able to use this one at the start of a pipeline like this, but it doesn't work yet:
```fish
> build/clang/debug/bin/vast exec 'shell "journalctl -e -o json" | parse json | head 1 | write csv to stdout'
!! invalid_argument: cannot execute pipeline that is not closed: shell "journalctl -e -o json" | parse json | head 1 | local print csv | save stdout
```

Reply:
You can add a `load file -` for now before it, but that doesn't quite do the correct thing because you don't want to open stdin. The operator instead needs an additional overload for when its input is void.

Reply:
Let me try make this happen as part of this PR.

Reply:
@tobim can you try again? I now made `shell` a source as well.

Reply:
Nice, it seems to be working!